python/GafferSceneTest/BoundQueryTest.py

```diff
@@ -0,0 +1,486 @@
+##########################################################################
+#
+#  Copyright (c) 2021, Cinesite VFX Ltd. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import unittest
+import imath
+
+import Gaffer
+import GafferScene
+import GafferSceneTest
+
+def randomName( gen, mnc, mxc ):
+
+	from string import ascii_lowercase
+
+	return ''.join( gen.choice( ascii_lowercase )
+		for _ in range( gen.randrange( mnc, mxc ) ) )
+
+class BoundQueryTest( GafferSceneTest.SceneTestCase ):
+
+	def testDefault( self ):
+
+		v = imath.V3f( 0.0, 0.0, 0.0 )
+		b = imath.Box3f( v, v )
+
+		bq = GafferScene.BoundQuery()
+
+		self.assertEqual( bq["__internalBound"].getValue(), b )
+		self.assertEqual( bq["bound"]["min"].getValue(), v )
+		self.assertEqual( bq["bound"]["max"].getValue(), v )
+		self.assertEqual( bq["center"].getValue(), v )
+		self.assertEqual( bq["size"].getValue(), v )
+
+	def testSpaceLocal( self ):
+
+		from random import Random
+		from datetime import datetime
+
+		r = Random( datetime.now() )
```

Comment:
As we chatted about this morning, I don't think it's usually a good idea to intentionally introduce non-determinism in the tests, since we don't want tests that fail in CI and can't be reproduced whenever we can avoid it.  If you think that generate random values could uncover an edge case, maybe generate 100 random values with a deterministic seed?